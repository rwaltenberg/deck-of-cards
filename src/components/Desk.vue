<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import DeckCard from './Card.vue'
import { StringToCard } from '../utils/card-conversion'

@Component({
  components: {
    DeckCard
  }
})
export default class Desk extends Vue {
  @Prop({ type: Array, default: () => [] })
  private value!: (Card | string)[]

  get cards () {
    return this.value.map(card => Array.isArray(card) ? card : StringToCard(card))
  }
}
</script>

<template>
  <ul class="desk">
    <deck-card v-for="card in cards" :key="card.join('')" :value="card" tag="li" />
  </ul>
</template>

<style lang="scss" scoped>
  .desk {
    align-items: flex-start;
    background-color: #d3d3d3;
    border: 2px solid #888888;
    box-sizing: border-box;
    display: flex;
    flex-wrap: wrap;
    min-height: 580px;
    list-style: none;
    margin: 0;
    padding: 0 25px;
    width: 1000px;

    .card {
      flex: 0 0 auto;
      margin: 24px;
    }
  }
</style>
