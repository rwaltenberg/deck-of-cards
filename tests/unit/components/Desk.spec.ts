import Desk from '@/components/Desk.vue'
import { shallowMount } from '@vue/test-utils'

describe('Desk.vue', () => {
  describe('with string type values', () => {
    it('renders properly', () => {
      const wrapper = shallowMount(Desk, {
        propsData: {
          value: ['AS', 'AC', 'AH', 'AD']
        }
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with Desk type values', () => {
    it('renders properly', () => {
      const wrapper = shallowMount(Desk, {
        propsData: {
          value: [['A', 'S'], ['A', 'C'], ['A', 'H'], ['A', 'D']]
        }
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('with empty value', () => {
    it('renders properly', () => {
      const wrapper = shallowMount(Desk)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
