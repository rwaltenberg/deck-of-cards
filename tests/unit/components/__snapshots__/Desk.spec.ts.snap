// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desk.vue with Desk type values renders properly 1`] = `
<ul class="desk">
  <deck-card-stub value="A,S" tag="li"></deck-card-stub>
  <deck-card-stub value="A,C" tag="li"></deck-card-stub>
  <deck-card-stub value="A,H" tag="li"></deck-card-stub>
  <deck-card-stub value="A,D" tag="li"></deck-card-stub>
</ul>
`;

exports[`Desk.vue with empty value renders properly 1`] = `<ul class="desk"></ul>`;

exports[`Desk.vue with string type values renders properly 1`] = `
<ul class="desk">
  <deck-card-stub value="A,S" tag="li"></deck-card-stub>
  <deck-card-stub value="A,C" tag="li"></deck-card-stub>
  <deck-card-stub value="A,H" tag="li"></deck-card-stub>
  <deck-card-stub value="A,D" tag="li"></deck-card-stub>
</ul>
`;
